File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        
        int maxlen = 0;
        if(nums.size() == 0)
            return 0;
        
        unordered_set<int> st;
        for(int ele: nums){
            st.insert(ele);
        }
        
        for(auto ele:st){
            //when element is starting of the sequence in set, calculate the length of sequence
            if(st.find(ele - 1) == st.end()){
                int length = 1;
                while(st.find(ele + 1) != st.end()){
                    length++;
                    ele = ele + 1;
                }
                maxlen = max(maxlen,length);
            }
        }
        return maxlen;
        
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 0:
            return 0
            
        nums.sort()
        print(nums)
        curlen,maxlen = 1,1

        for i in range(len(nums) - 1):
            if nums[i] == nums[i+1]:
                continue
            elif nums[i] == nums[i+1] - 1:
                curlen += 1
            else:
                curlen = 1
            maxlen = max(maxlen, curlen)
        
        return maxlen
        